source/loader/ze_loader.cpp

```diff
@@ -211,6 +211,10 @@ namespace loader
 
         // Use the previously init ddi table pointer to zeInit to allow for intercept of the zeInit calls
         ze_result_t res = globalInitStored->pfnInit(flags);
```

Comment:
In my testing with the null driver, this `pfnInit` points to `zeInit` in libze_null.so. How is `driver.initStatus` being set? Why are we retrieving `pfnInit` from the driver in the rest of this function only to eventually call and return an init function we were passed in? Won't `globalInitStored->pfnInit` be the same for every driver?

Reply:
When the intercept layer is set, globalInitStored->pfnInit here points to the intercept layer.

If you set ZE_ENABLE_LOADER_INTERCEPT=1 or have more than one driver, then zeInit() calls all the driver handles, not just this 1 driver.

In the ldrddi zeInit(), if at least one driver succeeds, then this passes.

driver.initStatus is default set to ZE_RESULT_SUCCESS. In the intercept layer, the driver.initStatus is set to an error if the ddi table get or zeInit() calls fail.

This check ensures that the correct driver status is returned.